# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditSubActionHandler.cpp
# Commit: d554ca7850eb
# Full Hash: d554ca7850eb362a0482c9f4c19caf288e65efb1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 0: Add automated tests of preserving selection while handling some commands r=m_kato
#   
#   The following patches touch the logic to restore selection after handling
#   commands.  However, it seems that they are not tested because I found some
#   regressions by manual testing, but didn't cause orange on tryserver.
# ==============================================================================

diff -r 86da8f2a28ea -r d554ca7850eb editor/libeditor/HTMLEditSubActionHandler.cpp
--- a/editor/libeditor/HTMLEditSubActionHandler.cpp	Wed Aug 03 23:58:51 2022 +0000
+++ b/editor/libeditor/HTMLEditSubActionHandler.cpp	Thu Aug 04 00:21:53 2022 +0000
@@ -5191,7 +5191,8 @@
       // If we've already found an ancestor block element indented, we need to
       // split it and remove the block element first.
       if (indentedParentElement) {
-        MOZ_ASSERT(indentedParentElement == content);
+        NS_WARNING_ASSERTION(indentedParentElement == content,
+                             "Indented parent element is not the <blockquote>");
         SplitRangeOffFromNodeResult outdentResult = OutdentPartOfBlock(
             *indentedParentElement, *firstContentToBeOutdented,
             *lastContentToBeOutdented, indentedParentIndentedWith);
@@ -10028,8 +10029,6 @@
 
 EditActionResult HTMLEditor::SetSelectionToAbsoluteAsSubAction(
     const Element& aEditingHost) {
-  MOZ_ASSERT(IsTopLevelEditSubActionDataAvailable());
-
   AutoPlaceholderBatch treatAsOneTransaction(
       *this, ScrollSelectionIntoView::Yes, __FUNCTION__);
   IgnoredErrorResult ignoredError;